W-23544053 Update policies-pdk-configure-features-headers-stop.adoc (#1120)
mulesoft/docs-gateway: pdk/1.9/modules/ROOT/pages/policies-pdk-configure-features-headers-stop.adoc

Use the Omni Gateway Policy Development Kit (PDK) `enable_stop_iteration` feature to simultaneously read and modify headers and body content.

> Without stop iteration, a policy sends headers to the upstream before sending the body. Use stop iteration if you must process the body before the headers are sent.

Reviewer: ```suggestion
Without stop iteration, a policy forwards the headers before the body. In the request flow, the request headers are sent to the upstream service before the request body; in the response flow, the response headers are sent to the downstream client before the response body.

Use stop iteration if you must process the body before the headers are sent.
```

> Use stop iteration if you must process the body before the headers are sent. Without stop iteration, a policy forwards the headers before the body. In the request flow, the policy sends request headers to the upstream service before the request body, In the response flow, The policy sends response headers to the client before the response body.

Reviewer: ```suggestion
Use stop iteration if you must process the body before the headers are sent. Without stop iteration, a policy forwards the headers before the body. In the request flow, the policy sends request headers to the upstream service before the request body, In the response flow, the policy sends response headers to the client before the response body.
```

Stop iteration is supported for Omni Gateway v1.11.14 and later.

> NOTE: The HTTP status code cannot be changed in the response filter when using stop iteration.

Reviewer: ```suggestion
NOTE: Under normal buffering, you can change the HTTP status code in the response filter when using stop iteration. However, if the buffered body exceeds `FLEX_DOWNSTREAM_CONNECTION_BUFFER_LIMIT_BYTES`, the filter loses control and the status code can no longer be changed.
```

> Stop iteration is supported for Omni Gateway v1.11.14 and later.

Reviewer: ```suggestion
Stop iteration is supported for Omni Gateway v1.11.4 and later.
```

> NOTE: If the buffered body exceeds the `FLEX_DOWNSTREAM_CONNECTION_BUFFER_LIMIT_BYTES`, the request is terminated with a `500` error code.

Reviewer: I'd mention here something related to the following, as we need to contrast both behaviours: _Under normal buffering, you can change the HTTP status code in the response filter when using stop iteration._

== Body Buffering and Size Limitations

> When a PDK policy transitions to the headers-body state with into_headers_body_state(), Flex Gateway buffers the entire body for both the request or the response before you policy runs. The policy doesn't receive the headers-body state until the full body has been buffered.

Reviewer: ```suggestion
When a PDK policy transitions to the headers-body state with into_headers_body_state(), Flex Gateway buffers the entire body for both the request and the response before your policy runs. The policy doesn't receive the headers-body state until the full body has been buffered.
```
